libcontainerd/client_daemon.go

```diff
@@ -139,12 +139,13 @@ func (c *client) Restore(ctx context.Context, id string, attachStdio StdioCallba
 	}()
 
 	t, err := ctr.Task(ctx, func(fifos *cio.FIFOSet) (cio.IO, error) {
-		rio, err = cio.NewDirectIO(ctx, fifos)
+		io, err := cio.NewDirectIO(ctx, fifos)
 		if err != nil {
 			return nil, err
 		}
 
-		return attachStdio(rio)
+		rio, err = attachStdio(io)
+		return rio, err
```

Comment:
I noticed the `rio, err` assignment was removed in an earlier commit, and added again here.

IIUC; it's needed for the `defer()`. Should we add a comment to prevent people from optimising away the assignment?

Also wondering now that two variables are used (`io`, `rio`) if a situation is possible where `rio` is nil, but `io` is not (and if `io` has to be closed inside the `defer()` in that case)

Reply:
Ya... it is needed for the defer, which is a problem. 

A comment might help, but I would rather fix the code to remove the problem. I think I have a fix.

> Also wondering now that two variables are used (io, rio) if a situation is possible where rio is nil, but io is not (and if io has to be closed inside the defer() in that case)

That shouldn't be a problem right now, but that is definitely a concern, so I'll change it to work with io in the defer.